### ISSUE TITLE
(Plug.Conn.NotSentError) a response was neither set nor sent from the connection

### ISSUE THREAD

**Issue by monukbizsoft** [mozilla/reticulum#669]
Hi 

When I merge the code into my local repo, and push it to the server. Then I have restarted reticulum and got the below error:

```
[info] GET /
[error] #PID<0.4309.0> running RetWeb.Endpoint (connection #PID<0.4305.0>, stream id 1) terminated
Server: example.com (https)
Request: GET /
** (exit) an exception was raised:
    ** (Plug.Conn.NotSentError) a response was neither set nor sent from the connection
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:67: Phoenix.Endpoint.Cowboy2Handler.maybe_send/2
        (phoenix) lib/phoenix/endpoint/cowboy2_handler.ex:43: Phoenix.Endpoint.Cowboy2Handler.init/4
        (cowboy) /hubs/hubs-actions-runner/reticulum/_work/reticulum/reticulum/deps/cowboy/src/cowboy_handler.erl:37: :cowboy_handler.execute/2
        (cowboy) /hubs/hubs-actions-runner/reticulum/_work/reticulum/reticulum/deps/cowboy/src/cowboy_stream_h.erl:306: :cowboy_stream_h.execute/3
        (cowboy) /hubs/hubs-actions-runner/reticulum/_work/reticulum/reticulum/deps/cowboy/src/cowboy_stream_h.erl:295: :cowboy_stream_h.request_process/3
        (stdlib) proc_lib.erl:249: :proc_lib.init_p_do_apply/3

```

Please check the screenshots

![Screenshot at Feb 02 13-45-41](https://user-images.githubusercontent.com/119565686/216268777-80d8c7ad-768f-4cfe-8eed-ff1a0a2aa8c2.png)

https://example.com/health is giving me 200 response code but when I try to access home page or internal page then it gave me error.

I have added .asdf/shims in first possition into $PATH file
```
**echo $PATH**
/root/.asdf/shims:/root/.asdf/bin:/root/.nvm/versions/node/v16.18.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
```

I am using below version of elixir and erlang
```
**# asdf current**
elixir          1.12.3          /hubs/hubs-actions-runner/reticulum/_work/reticulum/reticulum/.tool-versions
erlang          23.3            /hubs/hubs-actions-runner/reticulum/_work/reticulum/reticulum/.tool-versions
```

Thanks

**Comment by monukbizsoft** [comment]
Greetings!

I just rolled back last week's code on the server. and it is working fine now. So there is no need to consider this issue. Please reopen this if anyone is facing similar issue.

For now I'm going to close it.

Thanks

**Comment by albirrkarim** [comment]
i have that problem too with the latest reticulum code on 7 April 2023. please reopen this issue again

**Comment by albirrkarim** [comment]
I have solve this issue

<img width="676" alt="image" src="https://user-images.githubusercontent.com/29292018/230770782-870d62f7-2c15-4634-9e47-01b608969efa.png">

In my case, it is because. I set 

host=example.com
cors_proxy_host = host
assets_host = host 
link_host= host

please @monukbizsoft  close again this issue

**Comment by monukbizsoft** [comment]
Hi @albirrkarim 

yes you are absolutely right

This problem is caused by same url for host and asset_host.

Before setup this, I have tried to setup hub on digital ocean and it wants two different url for host and assets.
But I'm new to hubs so I was not sure about it and I did the same error while this repo on VPS server.

My setup on VPS has been working fine for two months without any issues

Thanks @albirrkarim for the VPS setup tutorial https://github.com/albirrkarim/mozilla-hubs-installation-detailed/blob/main/VPS_FOR_HUBS.md.

I have only one issue for now in reticulum and I have created another ticket for it https://github.com/mozilla/reticulum/issues/672

Thankyou
